canonical_facts.go

```diff
@@ -120,9 +120,14 @@ func GetCanonicalFacts() (*CanonicalFacts, error) {
 		return nil, err
 	}
 
-	facts.BIOSUUID, err = readFile("/sys/devices/virtual/dmi/id/product_uuid")
-	if err != nil {
-		return nil, err
+	if _, err := os.Stat("/sys/devices/virtual/dmi/id/product_uuid"); !os.IsNotExist(err) {
+		BIOSUUID, err := readFile("/sys/devices/virtual/dmi/id/product_uuid")
+		if err != nil {
+			return nil, err
+		}
+		facts.BIOSUUID = BIOSUUID
+	} else {
+		facts.BIOSUUID = ""
```

Comment:
Fun Go tidbit here. The `else` clause is unnecessary. Go initializes all variables to their "zero" value by default, so `facts.BIOSUUID` is already the zero value of `string` (which is an empty string `""`).

Reply:
I knew Go put all values to the zero value but I completely blanked when I was writing this... still learning! Thanks!